sqlmesh/core/config/connection.py

```diff
@@ -269,17 +269,17 @@ def init(cursor: duckdb.DuckDBPyConnection) -> None:
                         query = f"ATTACH '{path_options}'"
                         if not path_options.startswith("md:"):
                             query += f" AS {alias}"
-                        elif self.token:
-                            query += f"?motherduck_token={self.token}"
                     cursor.execute(query)
                 except BinderException as e:
                     # If a user tries to create a catalog pointing at `:memory:` and with the name `memory`
                     # then we don't want to raise since this happens by default. They are just doing this to
                     # set it as the default catalog.
+                    # If a user tried to attach a MotherDuck database/share which has already by attached via
```

Comment:
```suggestion
                    # If a user tried to attach a MotherDuck database/share which has already been attached via
```

Reply:
Why is this not an error? Why would anyone specify the same MD catalog more than once?

Reply:
So the way the implementation for MotherDuckConnectionConfig works right now, the db adapter uses `md:` as the connection string for duckdb, implicitly adding all the databases/shares available to the token account. if you then issue an `ATTACH` command to a specific db `md:some_db`, duckdb will emit an error.

per my comment (https://github.com/TobikoData/sqlmesh/pull/4001?notification_referrer_id=NT_kwDOADILGbMxNTMxOTIzNzM0MTozMjc5NjQx#discussion_r1996645657) above, we could either skip the ATTACH trusting that the db actually exists on the account, or we could check that it's there.

Reply:
I see. This makes sense, thank you